tests/models/test_dag.py

```diff
@@ -1754,3 +1757,98 @@ def test_count_number_queries(self, tasks_count):
                 state=State.RUNNING,
                 execution_date=TEST_DATE,
             )
+
+
+class TestDagDecorator:
+    DEFAULT_ARGS = {
+        "owner": "test",
+        "depends_on_past": True,
+        "start_date": timezone.utcnow(),
+        "retries": 1,
+        "retry_delay": timedelta(minutes=1),
+    }
+    DEFAULT_DATE = timezone.datetime(2016, 1, 1)
+    VALUE = 42
+
+    def setUp(self):
+        super().setUp()
+        self.operator = None
+
+    def tearDown(self):
+        super().tearDown()
+        clear_db_runs()
+
+    def test_set_dag_id(self):
+        @dag_decorator('test', default_args=self.DEFAULT_ARGS)
+        def noop_pipeline():
+            @task
+            def return_num(num):
+                return num
+
+            return_num(4)
+
+        assert noop_pipeline().dag_id, 'test'
```

Comment:
Should we add test that asserts that the returned value is a `DAG`? This will increase understanding imho